### ISSUE TITLE
show() to an IOBuffer with marker_space = :data results in an empty PDF

### ISSUE THREAD

**Issue by bjarthur** [MakieOrg/Makie.jl#4609]
following this [discussion](https://discourse.julialang.org/t/correct-way-to-save-makie-plot-to-iobuffer/87793) on how to save a figure to an IOBuffer, i have found that setting `markerspace` to `:data` results in the marks being absent from the resulting PDF.  with `markerspace` unspecified, everything works.

MWE:

```
using WGLMakie, CairoMakie, Base64, Bonito 

app = App() do
    f = Figure()   
    a, s = scatter(f[1,1], Point3f.(rand(10), rand(10), rand(10)),
                   markersize = 0.03, markerspace = :data     # delete this line and it will work
                   )

    CairoMakie.activate!() 
    io = IOBuffer()
    show(io, MIME"application/pdf"(), f)
    
    io2 = IOBuffer()
    print(io2, "data:application/pdf;base64,")
    print(io2, base64encode(take!(io)))

    DOM.div(f,
            "Click to download ",
            DOM.a("scatter.pdf",
                  href=String(take!(io2)),
                  download="scatter.pdf"))
end 
```

screenshot of GUI:
![Image](https://github.com/user-attachments/assets/80fd24d7-236b-4ee1-b99c-4c699b6f4563)

screenshot of PDF:
![Image](https://github.com/user-attachments/assets/42c86f0f-9fde-4b2d-b2d7-295c98d58002)

the latest version(s) of Makie freshly installed into a temporary environment on an apple M2 max:

```
(jl_Us6BFN) pkg> st
Status `/private/var/folders/s5/8d629n5d7nsf37f60_91wzr40000gq/T/jl_Us6BFN/Project.toml`
  [824d6782] Bonito v4.0.0
  [13f3f980] CairoMakie v0.12.16
  [276b4fcb] WGLMakie v0.10.16
  [2a0f44e3] Base64 v1.11.0
```

**Comment by bjarthur** [comment]
correct, `save` works fine.

**Comment by ffreyer** [comment]
This has nothing to do with WGLMakie. Will be fixed by #4663
